src/main/java/Entities/Item.java

```diff
@@ -188,7 +188,24 @@ public void setPriceHistoryData(ArrayList<Double> updatedPrices){
     public void setReviewCount(int reviewCount) { this.reviewCount = reviewCount;}
     public int getReviewCount() { return reviewCount;}
     public void setProductPrice(double newPrice) {this.itemPrice = newPrice;}
-    public void setProductCurrency(String newCurrency) {this.itemCurrency = newCurrency;}
+    public void setProductCurrency(String newCurrency) {
```

Comment:
Why are we creating new logic here? The currency use case exists to adhere to clean architecture with logic already in it.